packages/orchestration/src/util/module-config.ts

```diff
@@ -132,8 +135,32 @@ export function constructCompletionPostRequest(
   stream?: boolean,
   streamOptions?: StreamOptions
 ): CompletionPostRequest {
+
+  // templating cannot be string here.
+  let templatingConfig = config.templating as TemplatingModuleConfig;
+
+if (isTemplate(templatingConfig)) {
+
+  // Throw if the template is empty
+  if (!Array.isArray(templatingConfig.template) || templatingConfig.template.length === 0) {
+    throw new Error(
+      'Templating config was provided with an empty template. Either provide a valid template or omit the field entirely.'
+    );
+  }
+
+  // No change needed — templating is already valid
+} else if (!templatingConfig) {
```

Comment:
I feel like these if conditions together with the three-way comparison later in line 163 can be improved? The current logic feels too complicated.

Reply:
I will refactor this. This was a first draft anyway since I wanted to first finalize the API design.